### PR TITLE
Fix misused `aria-label` on mobile card table

### LINKED ISSUE #4131: A11Y: Mobile card variant of table misuses aria-label
In mobile card table pattern we use `aria-label` to show column headers in mobile view. This causes screen reader to only read column title and completely skip cell value.

We should not use `aria-label` this way.

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

- Updated `aria-label` to `data-heading`
- Updated docs to reflect change

Fixes
#4131 

## QA

- Open [demo](insert-demo-url)
- Turn voice over on while reviewing the table, ensure the content of the cells is read out not just the title. Currently the title is read out twice, not sure how to fix this, I tried to use `content: attr(data-heading) / ' ';` but this hides the title both times. 
- Review updated documentation:
  - /docs/base/tables#responsive

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

**Comment by bartaz** [comment]
@bethcollins92 thanks! Add this to upgrade guide as well, this is a significant breaking change that devs will need to carefully upgrade.

**Review by anthonydillon — APPROVED** [review]
LGTM :+1: did you want to put an a11y label on this PR?

**Comment by bethcollins92** [comment]
@bartaz Added - could you proof read it please?

File: templates/docs/examples/patterns/tables/table-mobile-card.html [hunk 1]
```diff
@@ -49,16 +49,16 @@
...
-      <td class="has-overflow" aria-label="Actions">
+      <td data-heading="FQDN">upward-muskox</td>
+      <td data-heading="Power">Off</td>
+      <td data-heading="Status">Allocated</td>
+      <td data-heading="Owner">sparkiegeek</td>
+      <td data-heading="Zone">london</td>
+      <td class="u-align--right"  data-heading="Cores">6</td>
+      <td class="u-align--right"  data-heading="RAM">4 GiB</td>
+      <td class="u-align--right"  data-heading="Disks">1</td>
+      <td class="u-align--right"  data-heading="Storage this is long enough to cause wrapping">500GB</td>
```
Comment (bartaz) [inline templates/docs/examples/patterns/tables/table-mobile-card.html:60]: It seems we no longer wrap these headings, so this label doesn't make sense anymore

```suggestion
      <td class="u-align--right"  data-heading="Storage">500GB</td>
```

File: templates/docs/examples/templates/responsive.html [hunk 2]
```diff
@@ -271,16 +271,16 @@ <h4>Table mobile card</h4>
...
-          <td class="has-overflow" aria-label="Actions">
+          <td data-heading="FQDN">upward-muskox</td>
+          <td data-heading="Power">Off</td>
+          <td data-heading="Status">Allocated</td>
+          <td data-heading="Owner">sparkiegeek</td>
+          <td data-heading="Zone">london</td>
+          <td class="u-align--right"  data-heading="Cores">6</td>
+          <td class="u-align--right"  data-heading="RAM">4 GiB</td>
+          <td class="u-align--right"  data-heading="Disks">1</td>
+          <td class="u-align--right"  data-heading="Storage this is long enough to cause wrapping">500GB</td>
```
Comment (bartaz) [inline templates/docs/examples/templates/responsive.html:282]:
```suggestion
          <td class="u-align--right"  data-heading="Storage">500GB</td>
```

**Pull request merged**